karapace/schema_reader.py

```diff
@@ -448,7 +451,10 @@ def _handle_msg_schema(self, key: dict, value: Optional[dict]) -> None:
                 return
         elif schema_type_parsed == SchemaType.PROTOBUF:
             try:
-                parsed_schema = ValidatedTypedSchema.parse(SchemaType.PROTOBUF, schema_str)
+                resolved_references, resolved_dependencies = self.resolve_schema_references(value)
+                parsed_schema = ValidatedTypedSchema.parse(
+                    SchemaType.PROTOBUF, schema_str, resolved_references, resolved_dependencies
+                )
```

Comment:
When the schema message is a delete there are no references in the data and resolved references and dependencies are `None`.
The parsing will fail as the data in this case will have unresolved dependency.

Register following schemas to subject one and subject two.
```
------------------------------------
Schema to be referenced
------------------------------------
syntax = "proto3";

message NoReference {
  string name = 1;
}

------------------------------------
Schema with reference to subject one schema NoReference.
------------------------------------
syntax = "proto3";

import "NoReference.proto";

message WithReference {
  string name = 1;
  NoReference ref = 2;
}
```

Try to delete the schema from subject two, `DELETE /subjects/subject_two/versions/1`. API returns 200, but deletion fails internally in schema reader, can be seen from logs.

Reply:
fixed